tapdb/assets_store.go

```diff
@@ -826,6 +827,24 @@ func (a *AssetStore) dbAssetsToChainAssets(dbAssets []ConfirmedAsset,
 				"internal key: %w", err)
 		}
 
+		// Fetch proof to obtain the block timestamp.
+		assetID := assetSprout.Genesis.ID()
+		locator := proof.Locator{
+			AssetID:   &assetID,
+			ScriptKey: *scriptKeyPub,
+			OutPoint:  &anchorOutpoint,
+		}
+		blob, err := a.FetchProof(ctx, locator)
+		if err != nil {
+			return nil, fmt.Errorf("unable to fetch proof for "+
+				"asset %s: %w", assetID, err)
+		}
+		p, err := blob.AsSingleProof()
```

Comment:
This should never work, as the proofs in the database are always proof files.

And anyway, this feels a bit much to do for each asset. I know Laolu mentioned this approach.
But my counter suggestion would be to add this block to `rpcServer.MarshalChainAsset()`:
```
	if a.AnchorBlockTimestamp == 0 && a.AnchorBlockHeight > 0 {
		a.AnchorBlockTimestamp = r.cfg.ChainBridge.GetBlockTimestamp(
			ctx, a.AnchorBlockHeight,
		)
	}
```

I know it's a bit of a hack. But it fixes the one call (from `fetchRpcAssets()`) that doesn't use `someProof.ToChainAsset()` (which sets the timestamp from the block header in the proof).